dbms/src/Flash/Pipeline/Schedule/Tasks/TaskHelper.h

```diff
@@ -40,6 +40,29 @@ namespace DB
 #define UNEXPECTED_STATUS(logger, status) \
     RUNTIME_ASSERT(false, (logger), "Unexpected task status {}", magic_enum::enum_name(status));
 
+#define FINALIZE_TASK(task) \
+    assert(task);           \
+    task->finalize();       \
+    task.reset();
+
+#define FINALIZE_TASKS(tasks) \
+    for (auto & task : tasks) \
+    {                         \
+        assert(task);         \
+        task->finalize();     \
+        task.reset();         \
+    }
+
+#define CATCH_AND_TERMINATE(log)                      \
+    catch (...)                                       \
+    {                                                 \
+        RUNTIME_ASSERT(                               \
```

Comment:
Why we have to terminate tiflash? Meaningless.

Reply:
If terminate is not called here, there will be a thread exit, some task/event in an unintended state, and possibly some exec_status.wait stuck.
I'm not sure if tiflash is still in a working state when just catching.